### PR TITLE
Support Jupyter Notebook 7

### LINKED ISSUE #9733: Support Jupyter Notebook 7

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #9733

### CODE REVIEW

File: pythonFiles/vscode_datascience_helpers/getServerInfo.py [hunk 1]
```diff
@@ -3,17 +3,20 @@
 import builtins
 
 try:
-    from notebook.notebookapp import list_running_servers
     import json
-
-    server_list = list_running_servers()
+    try:
+        from notebook.notebookapp import list_running_servers
```
Comment (DonJayamanne) [inline pythonFiles/vscode_datascience_helpers/getServerInfo.py:8]: Try jupyter 7, then fallback to latest version

File: pythonFiles/vscode_datascience_helpers/jupyter_daemon.py [hunk 2]
```diff
@@ -52,7 +52,10 @@ def m_exec_module(self, module_name, args=[], cwd=None, env=None):
                 return self._execute_and_capture_output(lambda: self._convert(args))
             if args[0] == "notebook" and args[1] == "--version":
                 try:
-                    from notebook import notebookapp as app
+                    try:
+                        from notebook import notebookapp as app
```
Comment (DonJayamanne) [inline pythonFiles/vscode_datascience_helpers/jupyter_daemon.py:56]: Try jupyter 7, then fallback to latest version

File: pythonFiles/vscode_datascience_helpers/jupyter_daemon.py [hunk 3]
```diff
@@ -151,7 +154,10 @@ def _convert(self, args):
             raise
 
     def _start_notebook(self, args, cwd, env):
-        from notebook import notebookapp as app
+        try:
+            from notebook import notebookapp as app
```
Comment (DonJayamanne) [inline pythonFiles/vscode_datascience_helpers/jupyter_daemon.py:158]: Try jupyter 7, then fallback to latest version

File: src/kernels/jupyter/launcher/jupyterConnection.node.ts [hunk 4]
```diff
@@ -149,9 +149,11 @@ export class JupyterConnectionWaiter implements IDisposable {
                 return;
             }
 
+            // For more recent versions of Jupyter the web pages are served from `/tree` and the api is at the root.
+            const pathName = url.pathname.endsWith('/tree') ? url.pathname.replace('/tree', '') : url.pathname;
```
Comment (DonJayamanne) [inline src/kernels/jupyter/launcher/jupyterConnection.node.ts:153]: Looks like we have a similar approach when working with jupyter hub as well, we strip out `tree`, as the ws & rest end points are at the root level.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**